src/main/java/de/uni_tuebingen/ub/nppm/db/QuelleDB.java

```diff
@@ -15,6 +17,48 @@ public static List getList() throws Exception {
         return getList(Quelle.class);
     }
 
+    public static List getList(Integer currentPage, Integer recordsPerPage, String filterTitle, String sort, String jumpToID) throws Exception {
+        Query query;
+        try (Session session = getSession()) {          
+            String q = "";
+            if(jumpToID != null && jumpToID.length() > 0){
+                q = "FROM Quelle q WHERE q.zuVeroeffentlichen = :zuV AND q.id = :id";
+                query = session.createQuery(q);
+                query.setParameter("id", Integer.valueOf(jumpToID));
+                query.setParameter("zuV", 1);
+                return query.list();
+            }else{
+                Integer start = null;
+                if(currentPage != null && recordsPerPage != null)
+                    start = currentPage * recordsPerPage - recordsPerPage;
+                if(sort.startsWith("title")){
+                    q = "FROM Quelle q WHERE q.zuVeroeffentlichen = :zuV AND q.bezeichnung LIKE :bez ORDER BY q.bezeichnung "+ (sort.equals("titleDown") ? "DESC" : "ASC");
+                }else if(sort.startsWith("belege")){
+                    q = "SELECT q FROM Quelle q LEFT JOIN q.einzelbelege e WHERE q.zuVeroeffentlichen = :zuV AND q.bezeichnung LIKE :bez " +
```

Comment:
This statement seems to be very slow, especially for belegeDown. Can this be optimized?

Reply:
ich habe es jetzt ohne join left und group by gemacht:

```
q = "SELECT * FROM quelle q WHERE q.zuVeroeffentlichen = :zuV AND q.bezeichnung LIKE :bez "
                            + "ORDER BY (SELECT COUNT(e.id) FROM einzelbeleg e WHERE e.QuelleID = q.id) " + (sort.equals("belegeDown") ? "DESC" : "ASC");
```